src/configure/clients/azureDevOpsClient.ts

```diff
@@ -0,0 +1,240 @@
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json;api-version=5.1-preview.1;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true"
+            },
+            method: "PATCH",
+            body: {
+                "allPipelines": {
+                    "authorized": true,
+                    "authorizedBy": null,
+                    "authorizedOn": null
+                },
+                "pipelines": null,
+                "resource": {
+                    "id": endpointId,
+                    "type": "endpoint"
+                }
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async createAndRunPipeline(pipelineConfiguration: WizardInputs): Promise<any> {
+        return await this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: Mustache.render("https://{{organizationName}}.visualstudio.com/_apis/Contribution/HierarchyQuery", pipelineConfiguration),
+            headers: {
+                "Accept": "application/json;api-version=5.0-preview.1;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true",
+                "Content-Type": "application/json"
+            },
+            method: "POST",
+            body: {
+                "contributionIds": [
+                    "ms.vss-build-web.create-and-run-pipeline-data-provider"
+                ],
+                "dataProviderContext": {
+                    "properties": {
+                        "connectionId": pipelineConfiguration.sourceProviderConnectionId, //GitHub endpoint id
+                        "sourceProvider": pipelineConfiguration.sourceRepositoryDetails.sourceProvider,
+                        "repositoryId": pipelineConfiguration.sourceRepositoryDetails.repositoryId,
+                        "repositoryName": pipelineConfiguration.sourceRepositoryDetails.repositoryName,
+                        "branch": pipelineConfiguration.sourceRepositoryDetails.branch,
+                        "sourceBranch": pipelineConfiguration.sourceRepositoryDetails.branch,
+                        "path": "./azure-pipelines.yml",
+                        "queue": "Hosted Ubuntu 1604",
+                        "commitId": pipelineConfiguration.sourceRepositoryDetails.commitId,
+                        "commitDescriptorName": "Set up CI/CD with Azure Pipelines",
+                        "sourcePage": {
+                            "routeValues": {
+                                "project": pipelineConfiguration.projectName
+                            }
+                        }
+                    }
+                }
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    private getConnectionData(): Promise<any> {
```

Comment:
you can start private methods with _ just for readability :)

Reply:
Thanks for suggesting that, but I'll stick to not using `_` for private properties. Also [Microsoft's Typescript coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines) say so :P